tide_site.services.yml

```diff
@@ -18,3 +18,8 @@ services:
     arguments: ['@entity_type.manager', '@entity.repository']
     calls:
       - [setContainer, ['@service_container']]
+  tide_site.event_subscriber:
+    class: Drupal\tide_site\EventSubscriber\TideSiteRouteAlter
+    arguments: ['@current_user']
```

Comment:
@vincent-gao I don't see you use the `current_user` service in the constructor of `TideSiteRouteAlter`. In fact your `TideSiteRouteAlter` class does not have a constructor.

Reply:
thanks. good point.  forgot to delete it.